src/hooks/useDisplayedWallets.ts

```diff
@@ -51,45 +60,60 @@ export const useDisplayedWallets = (): WalletInfo[] => {
           }) as WalletInfo
       );
 
-    // If Phantom wallet is detected, it must be in the 2nd slot.
-    // If there are no injected wallets, splice will just put it as the only item in the array.
-    if (phantomDetected) {
-      enabledInjectedWallets.splice(1, 0, {
-        connectorType: ConnectorType.PhantomSolana,
-        name: WalletType.Phantom,
-      });
-    }
+    const metamaskWallet = injectedMetaMask
+      ? {
+          connectorType: ConnectorType.Injected,
+          icon: injectedMetaMask.detail.info.icon,
+          name: injectedMetaMask.detail.info.name,
+          rdns: injectedMetaMask.detail.info.rdns,
+        }
+      : {
+          connectorType: ConnectorType.DownloadWallet,
+          name: WalletType.MetaMask,
+          downloadLink: METAMASK_DOWNLOAD_LINK,
+        };
+
```

Comment:
```suggestion
    const metamaskWallet = injectedMetaMask ?? {
          connectorType: ConnectorType.DownloadWallet,
          name: WalletType.MetaMask,
          downloadLink: METAMASK_DOWNLOAD_LINK,
        };

```

Reply:
`injectedMetaMask` isn't the right format D: it needs to go through the same transformation that we have in the `.map` function